### DISCUSSION TITLE
"Equal" attribution

### DISCUSSION THREAD

**Discussion by matthewlchambers** [USEPA/flowsa#267] [Q&A] [answered]
A clarification regarding attributing flows equally across multiple industries (used as a fallback method when other attribution sources are not available). If a flow of 6 is to be equally attributed to the NAICS codes 111001, 111002, 111003, and 211000, what is the desired final attribution?

- 111001: 1, 111002: 1, 111003: 1, and 211000: 3
- 111001: 1.5, 111002: 1.5, 111003: 1.5, and 211000: 1.5

I think the first attribution is preferable, but I'm not sure. Thoughts?

**Comment by catherinebirney** [comment, marked as answer]
My preference is also option 1 - it makes sense to equally attribute a parent naics to child naics, so equally attribute the hypothetical parent of 11 and 21 to 11 and 21 before further attributing to NAICS6. I initially attempted to equally attribute data as you've presented in option 1. However, some of the data we import into flowsa are presented in odd combinations of NAICS levels that I couldn't quite make option 1 work for. My function returned a dataset where the parent and summed child NAICS wouldn't be equal - so I defaulted to option 2. Take a look at  EIA_MECS_Energy - that is the dataset that caused problems for me.

**Reply by matthewlchambers** [reply]
Yes, I noticed that `EIA_MECS_Energy` was weird that way. But I wrote a pair of functions (in the course of adapting `GHG_national_2016_m1` to the recursive `.yaml` format) that I believe solve the issue:
https://github.com/USEPA/flowsa/blob/160536902933e21eaf933c8bf75d5f9fd20d967a/flowsa/data_source_scripts/temp_data_source_functions.py#L109-L230
since I prefer option 1 as well. This does cause some slight discrepancies in attribution between FBS's generated using the new code versus the old code. Are you comfortable with me ignoring those discrepancies, and continuing to have equal attribution proceed as in option 1?

**Reply by catherinebirney** [reply]
Oh fantastic - yes, we should use option 1 and ignore any discrepancies caused by the change in methodology. We will note the change in the new github release

**Comment by catherinebirney** [comment]
@matthewlchambers I'm looping back to this discussion as there is a need to equally attribute census of ag data from parent to child, where the child data isn't published (data contains NAICS4/5 and we want NAICS6). We don't have a general equal allocation function in the recursive branch yet, do we? I see the function to estimate suppressed data for qcew and the functions for the unique situation that is MECS.

**Reply by matthewlchambers** [reply]
The general equal allocation function is `equal_attribution()` in `flowby.py`. It assumes that mapping to sectors/industries has already occurred, then implements the algorithm discussed above.

**Reply by catherinebirney** [reply]
ah great, thanks!
